plonky2/src/gates/gate.rs

```diff
@@ -191,9 +198,19 @@ pub trait Gate<F: RichField + Extendable<D>, const D: usize>: 'static + Send + S
     }
 }
 
+pub trait AnyGate<F: RichField + Extendable<D>, const D: usize>: Gate<F, D> {
```

Comment:
This is just to allow gate type checking in `get_gate_tag_impl`, right? I think it'd be nice to have a comment explaining that